packages/gatsby/src/utils/tracer/jaeger-local.ts

```diff
@@ -0,0 +1,37 @@
+// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+// @ts-ignore
```

Comment:
Why there is `// @ts-ignore`?

Reply:
Apologies for the lack of comment. This is because we don't have a dependency on `jaeger-client` as this is an example to be used in other projects. No dependency means we get the `Cannot find module` error. 

Thinking about how the file is to be used, it might be a candidate for a `@ts-nocheck`, but if we do migrate it, and types are added, at least we'll get warnings if a jaeger change breaks it. I am now arguing myself in circles!

Reply:
We could try making `jaeger-client` optional peer dependency ( https://medium.com/@noamgjacobsonknzi/what-is-peerdependenciesmeta-acea887c32dd )

This is supported by `yarn@^1.13.0` and `npm@^6.11.0` - worry here is that `npm@^6.11.0` started shipping with Node.js in `Node.js 12.11.0`, so there would be quite a bit of not resolver peer deps warnings if user didn't upgrade npm if they are on earlier node versions

But also - I removed this `@ts-ignore` locally for testing and didn't see complaints when trying `eslint` and `typecheck` - what am I missing?

Reply:
Do you have `jaeger-client` installed at all? If I do `yarn why jaeger-client` I get the `We couldn't find a match` that's expected, then `yarn typecheck` throws an error when I run it. I also get the warning in VS Code, but it's probably safer to just check the official scripts.

Reply:
I also don't have it installed:
![Screenshot 2020-05-04 at 15 22 25](https://user-images.githubusercontent.com/419821/80970212-13764280-8e1b-11ea-98f8-c4e5d9b7103b.png)

Reply:
I am now in the same position as you! I noticed the last push failed typecheck on CI, which surprised me as there'd been no issues locally, so I did `yarn bootstrap; yarn typecheck` just to make sure I wasn't missing anything, and the CI error appeared. Fixed it, removed the ts-ignore in question, and now everything is perfectly happy. 

I wish I had a longer history set in my terminal so that I could screenshot it!

Reply:
Let's see what CI says after last commit you pushed. There just be more to it (or maybe it wasn't `typecheck` that failed?)

Reply:
The last CI failure was `typecheck`, but was to do with the extended interface I added. Because `initTracer` returns a `JaegerTracer` object, it was complaining that it didn't match my new interface. Obvious, really, but for some reason I didn't get the failure in VS Code. The `lint-staged` task doesn't trigger on changes to TypeScript files so it wasn't caught by husky.